tests/e2e/hci.go

```diff
@@ -210,87 +227,268 @@ var _ bool = ginkgo.Describe("hci", func() {
+
+		ginkgo.By("Verify PVs are accessible")
+		gomega.Expect(fss.CheckMount(ctx, client, statefulset, mountPath)).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By("Exit host from maintenance mode in remote cluster")
+		exitHostMM(ctx, targetHostSystem, mmStateChangeTimeout)
+		isHostInMaintenanceMode = false
+
+		ginkgo.By("Perform sts scale up and down and verify they are successful")
+		scaleUpStsAndVerifyPodMetadata(ctx, client, namespace, statefulset, replicas+1, true, true)
+		gomega.Expect(fss.CheckMount(ctx, client, statefulset, mountPath)).NotTo(gomega.HaveOccurred())
+		ssPods := fss.GetPodList(ctx, client, statefulset)
+		scaleDownStsAndVerifyPodMetadata(ctx, client, namespace, statefulset, ssPods, replicas-1, true, true)
+		gomega.Expect(fss.CheckMount(ctx, client, statefulset, mountPath)).NotTo(gomega.HaveOccurred())
+	})
+
+	/*
+		A host down in one AZ
+		steps:
+		1	Create an environment as described in the testbed layout above
+		2	Create statefulset2 with 1 replica
+		3	power off a host which has a k8s-worker with attached PVs in cluster1
+		4	wait for 5-10 mins, verify that the k8s-worker is restarted and brought up on another host
+		5	scale up statefulset2 and scale down statefulset1 to 2 replicas
+		6	power on the host used in step 3
+		7	reverse the operations done in step 5 and verify they are successful
+		8	cleanup all objects created during the test
+	*/
+	ginkgo.It("A host down in one AZ", ginkgo.Label(p0, block, vanilla, hci), func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		var err error
+		hostIp := ""
+
+		scParameters = map[string]string{}
+		scParameters[scParamStoragePolicyName] = remoteStoragePolicyName
+		scSpec := getVSphereStorageClassSpec(defaultNginxStorageClassName, scParameters, nil, "", "", false)
+		sc, err := client.StorageV1().StorageClasses().Create(ctx, scSpec, metav1.CreateOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		defer func() {
+			err := client.StorageV1().StorageClasses().Delete(ctx, sc.Name, *metav1.NewDeleteOptions(0))
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		ginkgo.By("Create 2 sts with 3 replica and 1 replica respectively")
+		sts1 := GetStatefulSetFromManifest(namespace)
+		sts2 := GetStatefulSetFromManifest(namespace)
+		var one int32 = 1
+		sts2.Spec.Replicas = &one
+		sts2.Name = "web-nginx"
+		sts2.Spec.Template.Labels["app"] = sts2.Name
+		sts2.Spec.Selector.MatchLabels["app"] = sts2.Name
+		framework.Logf("Creating statefulsets")
+		CreateStatefulSet(namespace, sts1, client)
+		CreateStatefulSet(namespace, sts2, client)
+		defer func() {
+			ginkgo.By(fmt.Sprintf("Deleting all statefulsets in namespace: %v", namespace))
+			fss.DeleteAllStatefulSets(ctx, client, namespace)
+		}()
+
+		replicas1 := *(sts1.Spec.Replicas)
+		replicas2 := *(sts2.Spec.Replicas)
+		// Waiting for pods status to be Ready
+
+		fss.WaitForStatusReadyReplicas(ctx, client, sts1, replicas1)
```

Comment:
try to use this method createStafeulSetAndVerifyPVAndPodNodeAffinty and scaleUpStatefulSetPod to handle sts creation along with scale up n down

Reply:
I have made change to use createCustomisedStatefulSets to create specific statefulsets instead of createStafeulSetAndVerifyPVAndPodNodeAffinty as it does not take care of sts name.